services/gps2tsdb/test/gps_verify/gps_verify.py

```diff
@@ -1,75 +1,104 @@
-time.sleep(10)
+points_received = 0
 
-print('Connecting to the database')
-connectionurl='postgresql://' + os.environ['db_user'] + ':' + os.environ['db_password'] + '@postgres:' + os.environ['db_port'] + '/' + os.environ['db_database']
-print('Initing Postgres obj')
-db = postgres.Postgres(url=connectionurl)
+async def run(loop):
+  
+  print('Connecting to the database')
+  connectionurl='postgresql://' + os.environ['db_user'] + ':' + os.environ['db_password'] + '@postgres:' + os.environ['db_port'] + '/' + os.environ['db_database']
 
-# This should be replaced query that deletes all lines from the device 'fake gps' or whatever but until
-# gps device is recorded this is what we got 
-print('Truncating GPS database')
-db.run('TRUNCATE gps;')
+  # Try to connect to the DB. It may be starting up so we should try a few timees
+  # before failing. Currently trying every second 60 times
+  tries = 0
+  maxtries = 60
+  sleeptime = 1
+  db_connected = False
+  while not db_connected:
+    try:
+      db = postgres.Postgres(url=connectionurl)
+    except OperationalError as e:
+      if tries < maxtries:
+        print("Database connection attempt", tries, "failed. Database may still be starting. Sleeping", sleeptime, "s and trying again")
+        print(e)
+        tries = tries + 1
+        time.sleep(sleeptime)
+      else:
+        print("FATAL: Could not connect to db after", tries, "tries. Exiting")
+        sys.exit(-1)
+    else:
+      print("DB successfully connected")
+      db_connected = True
 
-# Based on example from here: https://zguide.zeromq.org/docs/chapter2/#Node-Coordination
-print('Setting up zeromq')
-zmq_ctx = zmq.Context()
+  # Sleep for 2 seconds to allows gps2tsdb to create the gps table
+  time.sleep(5)
 
-# First, connect our subscriber socket
-print('Connecting to zmq sub socket')
-zmq_sub = zmq_ctx.socket(zmq.SUB)
-zmq_sub.connect('tcp://gps_replay:5561')
-zmq_sub.setsockopt(zmq.SUBSCRIBE, b'')
+  # This should be replaced query that deletes all lines from the device 'fake gps' or whatever but until
+  # gps device is recorded this is what we got 
+  print('Truncating GPS database')
+  db.run('TRUNCATE gps;')
+    
+  print('Setting up NATS')
+  # Connect to NATS server 
+  nc = NATS()
+  await nc.connect("nats://nats:4222")
+  
+  # Notifies the non-callback function that we are finished
+  fin = asyncio.Future() 
 
-time.sleep(1)
+  async def message_handler(msg): 
+    nats_gps_point = json.loads(msg.data.decode())
 
-# Second, synchronize with publisher
-print('Connecting to zmq sync socket')
-zmq_sync = zmq_ctx.socket(zmq.REQ)
-zmq_sync.connect('tcp://gps_replay:5562')
+    # Debugging
+    #print("NATS type:", type(nats_gps_point), "\nmessage:", nats_gps_point)
+    
+    # Check if this is a control message signifying the end of the transmission
+    if nats_gps_point == "END": 
+      print('End control message received. No more gps points to check')
+      nonlocal fin
+      fin.set_result("END")
```

Comment:
Maybe this is the way for python? I'm more used to message passing based flows, but that may not be "pythonic"?